bemserver_ui/internal_api/analysis/weather.py

```diff
@@ -0,0 +1,288 @@
+                    "data": {},
+                    "yAxis": 1,
+                    "timeseries": {
+                        "id": "",
+                        "name": "",
+                        "desc": "",
+                        "unit": "",
+                    },
+                },
+                "current": {
+                    "name": WeatherParameter.RELATIVE_HUMIDITY.value,
+                    "data": {},
+                    "yAxis": 1,
+                    "timeseries": {
+                        "id": "",
+                        "name": "",
+                        "desc": "",
+                        "unit": "",
+                    },
+                },
+            },
+        },
+        "Solar radiation": {
+            WeatherParameter.SURFACE_SOLAR_RADIATION.name: {
+                "forecast": {
+                    "name": WeatherParameter.SURFACE_SOLAR_RADIATION.value
+                    + " forecast",
+                    "data": {},
+                    "yAxis": 0,
+                    "timeseries": {
+                        "id": "",
+                        "name": "",
+                        "desc": "",
+                        "unit": "",
+                    },
+                },
+                "current": {
+                    "name": WeatherParameter.SURFACE_SOLAR_RADIATION.value,
+                    "data": {},
+                    "yAxis": 0,
+                    "timeseries": {
+                        "id": "",
+                        "name": "",
+                        "desc": "",
+                        "unit": "",
+                    },
+                },
+            },
+            WeatherParameter.DIRECT_NORMAL_SOLAR_RADIATION.name: {
+                "forecast": {
+                    "name": WeatherParameter.DIRECT_NORMAL_SOLAR_RADIATION.value
+                    + " forecast",
+                    "data": {},
+                    "yAxis": 0,
+                    "timeseries": {
+                        "id": "",
+                        "name": "",
+                        "desc": "",
+                        "unit": "",
+                    },
+                },
+                "current": {
+                    "name": WeatherParameter.DIRECT_NORMAL_SOLAR_RADIATION.value,
+                    "data": {},
+                    "yAxis": 0,
+                    "timeseries": {
+                        "id": "",
+                        "name": "",
+                        "desc": "",
+                        "unit": "",
+                    },
+                },
+            },
+        },
+        # Add other graphs here
+    }
+
+    resp = flask.g.api_client.weather_ts_by_sites.getall(
+        dt_start=dt_start, dt_end=dt_end, site_id=site_id
+    )
+
+    liste_ts_id = []
+    liste_forecast_ts_id = []
+
+    for data in resp.data:
+        for key in respData.keys():
+            if data["parameter"] in respData[key].keys():
+                if data["forecast"]:
```

Comment:
To avoid the repetition of code in the 2 cases (forecast or not), you should use a var that contains the key name "forecast" or "current" depending on the context of the data treated.
For example:
`weather_type = "forecast" if data["forecast"] else "current"`
and then just write once for the 2 cases
`respData[key][data["parameter"]][weather_type]["timeseries]["id"] = data["timeseries_id"]`
and so on...